joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/utils/PackageTable.scala

```diff
@@ -1,26 +1,49 @@
 package io.joern.rubysrc2cpg.utils
 
+import io.joern.rubysrc2cpg.astcreation.AstCreatorHelper
+import io.joern.x2cpg.Defines.DynamicCallUnknownFullName
+
+import java.util.concurrent.ConcurrentHashMap
 import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
 
 case class MethodTableModel(methodName: String, parentClassPath: String, classType: String)
 
 case class PackageContext(moduleName: String, packageTable: PackageTable)
 
-class PackageTable() {
+class PackageTable() extends AstCreatorHelper {
 
-  private val methodTableMap = mutable.HashMap[String, mutable.HashSet[MethodTableModel]]()
+  private val methodTableMap = new ConcurrentHashMap[String, mutable.HashSet[MethodTableModel]]()
 
   def addPackageMethod(moduleName: String, methodName: String, parentClassPath: String, classType: String): Unit = {
     val packageMethod = MethodTableModel(methodName, parentClassPath, classType)
-    methodTableMap.getOrElseUpdate(moduleName, mutable.HashSet.empty[MethodTableModel]) += packageMethod
+    val moduleMethodSet = methodTableMap.synchronized {
+      methodTableMap.computeIfAbsent(moduleName, _ => mutable.HashSet.empty[MethodTableModel])
+    }
+    moduleMethodSet.add(packageMethod)
   }
 
-  def getMethodFullNameUsingName(packageUsed: List[String], methodName: String): List[String] =
-    packageUsed
-      .filter(methodTableMap.contains)
-      .flatMap(module =>
-        methodTableMap(module)
+  def getMethodFullNameUsingName(packageUsed: List[String], methodName: String): Option[String] = {
+    if (isExcluded(methodName)) {
```

Comment:
The result of the if block is not used anywhere

Reply:
done.